### DISCUSSION TITLE
Struggling hours with: No profile for team 'TEAM_ID' matching 'Apple Distribution' found: Xcode couldn't find any provisioning profiles matching

### DISCUSSION THREAD

**Discussion by couchlab-pl** [codemagic-ci-cd/codemagic-docs#979] [Q&A] [answered]
HI,

Hope someone already solved it. I've an application for iOS generated via https://www.pwabuilder.com/ and I want to build and sign it in codemagic but stucked with that error. Certs are generated via Team Code signinig etc. 

thanks in advance
Martin

**Comment by icarusdust** [comment]
Hi @couchlab-pl, please share a build id as well, so that we can look into it in details. In general,  it could be because of targeting a different scheme if using flavors, certificate type could be unknown for profile type etc.

**Reply by couchlab-pl** [reply]
Thanks, build_id: 6330140e27eafd7386404b69

**Reply by couchlab-pl** [reply]
@icarusdust did you have a chance to look at it ?

**Reply by icarusdust** [reply]
@couchlab-pl I'll check it in a bit and keep you updated.

**Reply by icarusdust** [reply]
@couchlab-pl I see that you are working with code signing identities. It appears that you are not uploading the correct provisioning profile. Please make sure in your Apple dev account that you are downloading the correct .mobileprovision

**Reply by couchlab-pl** [reply]
Unfortunately, I already knew this from the error message. Certificates are generated by CodeMagic, profiles are valid - all attached. The question is what the correct profile should be - since CodeMagic says that they are all ok.
![image](https://user-images.githubusercontent.com/95092959/192528148-0fd4452d-eaff-47d7-ae08-21f0cf0ac001.png)

![image](https://user-images.githubusercontent.com/95092959/192528177-ca4e124d-aa64-44a7-84d7-79dfdf890d26.png)

**Reply by couchlab-pl** [reply]
@icarusdust any idea what might be wrong ?

**Reply by icarusdust** [reply]
@couchlab-pl let's get started with the following suggestions:
1. Remove `keychain initialize` and` keychain add-certificates` steps as this is already done as part of Set up code signing identities build step.
2. Add `--archive-flags='-destination generic/platform=iOS'` to `xcode-project build-ipa`  command.

**Reply by couchlab-pl** [reply]
Thanks a lot, work like a charm !

**Comment by icarusdust** [comment, marked as answer]
https://github.com/orgs/codemagic-ci-cd/discussions/979#discussioncomment-3759985
